pkgs/applications/misc/dmenu/default.nix

```diff
@@ -28,11 +29,15 @@ stdenv.mkDerivation rec {
   '';
 
   preConfigure = ''
-    sed -i "s@PREFIX = /usr/local@PREFIX = $out@g" config.mk
+    makeFlagsArray+=(
+      PREFIX="$out"
+      CC="$CC"
+      # default config.mk hardcodes dependent libraries and include paths
+      INCS="`$PKG_CONFIG --cflags fontconfig x11 xft xinerama`"
+      LIBS="`$PKG_CONFIG --libs   fontconfig x11 xft xinerama`"
+    )
   '';
```

Comment:
It might be possible to use a Nix `makeFlags` variable here with `PREFIX=$(out)`, `INCS=$(shell pkg-config …)`, etc., though it’s possible that would require `__structuredAttrs` because of the spaces.

Reply:
I agree it's not perfect. Are any of these patches in `nixpkgs` so we could test them? Or they are all external? (I would like to stare at the example modifications as well)

The `X11INC` and `X11LIBS` are not passed as is but are prefixed with `-I`/`-L`. That makes it awkward to just pass `pkg-config` resolution there:

```
$ cat config.mk
...
X11INC = /usr/X11R6/include
X11LIB = /usr/X11R6/lib
XINERAMALIBS  = -lXinerama
XINERAMAFLAGS = -DXINERAMA
FREETYPELIBS = -lfontconfig -lXft
FREETYPEINC = /usr/include/freetype2
INCS = -I$(X11INC) -I$(FREETYPEINC)
LIBS = -L$(X11LIB) -lX11 $(XINERAMALIBS) $(FREETYPELIBS)

CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"$(VERSION)\" $(XINERAMAFLAGS)
CFLAGS   = -std=c99 -pedantic -Wall -Os $(INCS) $(CPPFLAGS)
LDFLAGS  = $(LIBS)
```

We need a place to inject the include paths to `freetype`, `xinerama` and `xft` somewhere. Unfortunately none of the `config.mk` variables are great for that. I would expect that adding extra libraries in patches would still require you to pass header files explicitly via `makeFlags` or via `NIX_CFLAGS_COMPILE`. But `nixpkgs` clobber makes it even worse.

Reply:
> Are any of these patches in `nixpkgs` so we could test them? Or they are all external? (I would like to stare at the example modifications as well)

I don't think any are in nixpkgs directly, but plenty are [listed on dmenu's page](https://tools.suckless.org/dmenu/patches/).